lib/tbot/renew_test.go

```diff
@@ -162,3 +162,77 @@ func TestDatabaseRequest(t *testing.T) {
 	require.Equal(t, "baz", route.Username)
 	require.Equal(t, "mysql", route.Protocol)
 }
+
+func TestAppRequest(t *testing.T) {
+	t.Parallel()
+
+	const appName = "foo"
+	log := libutils.NewLoggerForTests()
+	// Make a new auth server.
+	fc, fds := testhelpers.DefaultConfig(t)
+	fc.Apps.Service = libconfig.Service{
+		EnabledFlag: "true",
+	}
+	fc.Apps.Apps = []*libconfig.App{
+		{
+			Name:       appName,
+			PublicAddr: "foo.example.com",
+			URI:        "http://foo.example.com:1234",
+			StaticLabels: map[string]string{
+				"env": "dev",
+			},
+		},
+	}
+	_ = testhelpers.MakeAndRunTestAuthServer(t, log, fc, fds)
+	rootClient := testhelpers.MakeDefaultAuthClient(t, log, fc)
+
+	// Wait for the app to become available. Sometimes this takes a bit
+	// of time in CI.
+	require.Eventually(t, func() bool {
+		_, err := getApp(context.Background(), rootClient, appName)
+		return err == nil
+	}, 5*time.Second, 100*time.Millisecond)
```

Comment:
Do you think 5s is enough time to avoid flakiness on a constrained builder?

Reply:
In my experience it does seem like a borderline race condition when this happens (at least for databases), but I'll increase it to 10s to be safe.